job_list_params.go

```diff
@@ -193,6 +190,23 @@ func (p *JobListParams) toDBParams() (*dblist.JobListParams, error) {
 		return nil, errors.New("invalid sort order")
 	}
 
+	if p.sortField == JobListOrderByFinalizedAt {
+		currentNonFinalizedStates := make([]rivertype.JobState, 0, len(p.states))
+		for _, state := range p.states {
+			//nolint:exhaustive
+			switch state {
+			case JobStateCancelled, JobStateCompleted, JobStateDiscarded:
+			default:
+				currentNonFinalizedStates = append(currentNonFinalizedStates, state)
+			}
+		}
+		// This indicates the user overrode the States list with only non-finalized
+		// states prior to then requesting FinalizedAt ordering.
+		if len(currentNonFinalizedStates) == 0 {
+			return nil, errors.New("cannot order by finalized_at with non-finalized state filters")
```

Comment:
Could we put the non-finalized states into the error message with a `%+v` or something? Just helps make the error quicker to resolve.